routers/api/v1/repo/branch.go

```diff
@@ -173,11 +173,35 @@ func CreateBranch(ctx *context.APIContext) {
 		return
 	}
 
-	if len(opt.OldBranchName) == 0 {
-		opt.OldBranchName = ctx.Repo.Repository.DefaultBranch
+	var oldCommit *git.Commit
+	var err error
+
+	if len(opt.OldRefName) > 0 {
```

Comment:
I think we need to make sure `OldRefName` and `OldBranchName` shouldn't appear at the same time.

Otherwise LGTM.